### PR TITLE
Improve javadoc on ModuleDependency

### PR DESCRIPTION
**Pull request by tresat**
Capture comments from https://gradle.slack.com/archives/C01D0BR067J/p1701245787426529

### CODE REVIEW

**Review by big-guy — APPROVED** [review]

**Comment by tresat** [comment]
@bot-gradle test and merge

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/ModuleDependency.java [hunk 1]
```diff
@@ -81,6 +86,14 @@ public interface ModuleDependency extends Dependency, HasConfigurableAttributes<
 
     /**
      * Returns the artifacts belonging to this dependency.
+     * <p>
+     * There will typically be a single artifact for a dependency, but this is not always true.  Modules can use
```
Comment (jvandort) [inline subprojects/core-api/src/main/java/org/gradle/api/artifacts/ModuleDependency.java:90]: > There will typically be a single artifact for a dependency

This wording here makes it seen each `ModuleDependency` will usually return a single artifact from this method. By default, `getArtifacts` will return an empty set unless the user adds an artifact themselves. 

I think we should revise this wording to be more clear here

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/DependencyArtifact.java [hunk 2]
```diff
@@ -18,8 +18,9 @@
 import javax.annotation.Nullable;
 
 /**
- * <p>An {@code Artifact} represents an artifact included in a {@link org.gradle.api.artifacts.Dependency}.</p>
- * An artifact is an (immutable) value object.
+ * Represents an artifact included in a {@link org.gradle.api.artifacts.Dependency}.
+ *
+ * @implSpec all implementations should be written as immutable value objects
```
[the remark this answers is not in the document]
Comment (tresat) [reply subprojects/core-api/src/main/java/org/gradle/api/artifacts/DependencyArtifact.java:23]: Hmmm, good point.  This was existing documentation I just moved to a more appropriate section.  I'll remove it.

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/ModuleDependency.java (hunk 1 above)
Comment (tresat) [reply subprojects/core-api/src/main/java/org/gradle/api/artifacts/ModuleDependency.java:90]: I'm all about adding more details to increase clarity.  I'll add that.

**Pull request merged**